### ISSUE TITLE
[SUGGESTION] Integrated Solution Explorer with exists Explorer  

### ISSUE THREAD

**Issue by mojtabakaviani** [microsoft/vscode-dotnettools#200]
### Describe the feature you'd like

I think better to add solution explorer capabilities to exists explorer instead of to created new solution explorer view, because exists explorer is project folder and support many project types, files and actions. with new solution explorer need movement between explorer and solution explorer many times and decrease productivity and performance of developers.

### Alternatives considered

Add solution explorer capabilities to exists explorer.

**Comment by mojtabakaviani** [comment]
Thanks @timheuer, I hope integrated to default Explorer view very soon.

**Comment by timheuer** [comment]
> Thanks @timheuer, I hope integrated to default Explorer view very soon.

Appreciate the enthusiasm.  This is not the direction we've set out, so setting expectations, it won't be very soon ;-)

**Comment by burtonrodman** [comment]
i too very much prefer the main Explorer view in vscode (folder view) and while its nice to have the Solution view, it would be most productive for me if the actions were simply integrated into the Explorer view.  as it is, I only activate the Solution Explorer when needing to do an operation on the sln file that I might have done at the CLI...  dotnet sln add, dotnet new, etc... (which I am perfectly comfortable with).

I might be inclined to switch to the Solution Explorer if all of the existing file operation commands were implemented... most notably Shft+F2 for "File: New File...", and there is a command to jump between folder view and solution view at the same logical location.

Perhaps instead of a separate node in the Explorer, it could be a toggle between folder view and solution view like VS.  I know this doesn't quite fit the UX of vscode, but just a thought.

just my 2 cents ;)

**Comment by mojtabakaviani** [comment]
Yes, a solution is to add switch button in default Explorer view. but my opinion is to auto detect dotnet projects and customize folders and add actions to context menu.

**Comment by timheuer** [comment]
> it could be a toggle between folder view and solution view like VS

Basically this is what it is today and there are a few ways to make it more of a 'toggle' rather than multiple views in the Explorer view panel.

1) Use the secondary side bar:
![image](https://github.com/microsoft/vscode-dotnettools/assets/4821/71df75b0-2d75-4d4b-9318-6c9ac7cda4d2)
This can be done by clicking the secondary side bar icon to expose it and then you can drag any explorer panel to this area

2) drag the solution view to the activity bar (vertical icons) and then you could have more of a toggle feel:
![2023-07-05_07-21-29](https://github.com/microsoft/vscode-dotnettools/assets/4821/42256d51-9ee6-4930-a36f-17235397d0a2)

Changing the default workspace view is not something we're looking to do honestly. For single-projects I can see how this makes sense but our intent was 'solution' views for multiple projects.  Additionally, when you have polyglot scenarios, we don't want to confuse the default explorer file view.  Exposing some commands (maybe like build, etc.) via right-click options on a .csproj might be simpler things we can look at, but fully taking what we have (and plan to expand) into the default view is not our plan right now.

**Comment by burtonrodman** [comment]
understood.  thanks for the suggestions

**Comment by attilah** [comment]
@timheuer I am happy that I'm not alone with my thoughts about the counterproductive aspect of having a separate view for Solution Explorer.

What do you think about just adding the functionality to the context menu of the workspaces view for the same nodes that Solution Explorer handles. Most of them opening up new dialogs, so that level of integration would be quite easy to add, you already know the node mapping between the two.

I don't expect to have a Dependencies node in the workspace view, but I do like to have `Add New Project`, `Add New File`, 'New Project', 'Add Existing Project', 'Run', 'Debug', etc there.

If project to project or nuget dependency management will be implemented by opening a new window/form/dialog, then it would be also an easy add.

With this approach we'd get the 90% of the most used functionality without using the Solution Explorer view.

**Comment by Sour-Codes** [comment]
I thought a bloat free, watered down, straight-to-the-point editor was great -- aka VS Code
Now, I see we're slowly trying to turn Code into Visual Studio, unfortunately.

One of the key features I fell in love with using Code over Studio is that you can do everything straight from the keyboard.  Mouse/Context menu to do things, such *Add project reference*, is slow, limited, and reminiscent of Visual Studio where one can just copy/paste `<ProjectReference />`.

Reading the comments, I'm all for *__NOT__* merging/replacing _Explorer_ with _Solution Explorer_ -- please keep these two forever apart or an ability to prefer one over the other.

I get that you guys are trying to cater to all audiences, but instead of trying to make VS Code look/work more like Visual Studio, why not just tell these individuals to stick to Visual Studio?

(I need to save a version of Code sooner rather than later before I no longer recognize it as VS Code with all these cosmetic, mouse-friendly changes.)

**Comment by attilah** [comment]
@Sour-Codes I guess you missed the post about killing VS for Mac ;-) So under non-windows MSFT does not provide an IDE experience anymore, so "we" can't stick to Visual Studio...

**Comment by Sour-Codes** [comment]
Ah, I did miss that, @attilah - I tend not to concern myself with things Apple, tbh.  It could be worse, like MSFT charging $99 annually just to code on non-MSFT tools to run things on MSFT, as I have to do with Apple/Developer account/iPad/etc... :[  Nothing is perfect...

Hmm, how do we solve giving Mac a Visual Studio IDE to replace Visual Studio for Mac while not turning VS Code into Visual Studio...?

I can't speak for others, but I've never looked back to Studio ever since Code was a thing and only needed it for things locked to Visual Studio (e.g.: Xamarin, MAUI debugging on devices, [.NET nanoFramework](https://www.nanoframework.net/) debugging, Unity/Unreal, UWP/HoloLens remote debugging, etc...)

---

I'm just hoping VS Code doesn't turn into the new, new Visual Studio.  Unless you want all the bells and whistles that comes with Visual Studio for Mac, what are you missing from VS Code?

**Comment by timheuer** [comment]
@Sour-Codes your comments are the reason why we are not integrating things into the core of Code. You are free to still copy/paste references, use keyboard shortcuts etc. all of the capabilities of anything we deliver do not impede those and are optional. You can hide panels in VS Code as well to not even show them.
